tools/links.py

```diff
@@ -122,23 +122,23 @@ def search_and_replace(link: str, new_link: str):
         The link to replace the existing one.
     """
     # Traverse the docs directory
-    for root, _, files in os.walk(DOCS_DIRECTORY):
+    for root, _, files in Path.walk(DOCS_DIRECTORY):
         # Skip the _static subdirectory
-        if "_static" in root.split(os.sep):
+        if "_static" in root.parts:
             continue
 
         # Process the files
         for file in files:
-            file_path = os.path.join(root, file)
-            with open(file_path, "r") as f:
-                content = f.read()
+            file_path = root / file
+            with Path.open(file_path, "r") as file:
+                content = file.read()

```

Comment:
Probably we could use:

```suggestion
            content = file_path.read_text(encoding="utf-8")

```